### DISCUSSION TITLE
Pydantic validation is executed multiple times if same model used as request body and in dependency

### DISCUSSION THREAD

**Discussion by tikon93** [fastapi/fastapi#8659] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI, Depends
from pydantic import BaseModel, StrictInt

app = FastAPI()


class Model(BaseModel):
    foo: StrictInt
    bar: StrictInt


def dependency(request: Model):
    return request.foo + request.bar


@app.post("/")
def foo(request: Model, processed_data: int = Depends(dependency)):
    return "hello"


uvicorn.run(app, port=8000)
```

### Description

I'm trying to make some request data postprocessing using FastAPI dependency.
It works fine, but seems like data from request is dumped into Pydantic model multiple times. It leads to some overhead and excessive validation errors if any. 

For example, following request:
`$ curl --location --request POST 'http://localhost:8000' --header 'Content-Type: application/json' --data-raw '{"bar": 2}'`
Will produce following error:
```
{
    "detail": [
        {
            "loc": [
                "body",
                "foo"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        },
        {
            "loc": [
                "body",
                "foo"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```

which looks weird.
Is there a correct way of doing this stuff using Pydantic models in methods declaration?

### Operating System

Linux

### FastAPI Version

0.68.1

### Python Version

3.9.5

**Comment by adamserafini** [comment]
I think the issue can be generalised even further (made an issue here: https://github.com/tiangolo/fastapi/issues/4122), perhaps my issue is actually a generalised duplicate of this one.

It seems if any dependency that results in validation appears X times in the dependency tree, any validation error for that dependency will appear X times in the error output.

If it's currently expected, I believe we could turn the issue into a feature request to 'deduplicate' these validation errors.

**Reply by escribano89** [reply]
Is there any update about this point?

**Reply by JavierSanchezCastro** [reply]
idk really, I opened a discussion some weeks ago about something I think is related: #9535

**Comment by mdczaplicki** [comment]
Anu update on that?

**Reply by Kludex** [reply]
This doesn't happen using Pydantic V2 anymore.

**Reply by JavierSanchezCastro** [reply]
Still happening:

<img width="613" alt="image" src="https://github.com/user-attachments/assets/20ebbe4f-9b92-45c9-adfb-dd43ddf569f5" />
